integration/terraform/ec2/win/main.tf

```diff
@@ -1,33 +1,71 @@
-resource "aws_instance" "integration-test" {
-  ami                    = data.aws_ami.latest.id
-  instance_type          = var.ec2_instance_type
-  key_name               = var.key_name
-  iam_instance_profile   = var.iam_instance_profile
-  vpc_security_group_ids = var.vpc_security_group_ids
-  get_password_data = true
+#Create a unique testing_id for each test
+resource "random_id" "testing_id" {
+  byte_length = 8
+}
+
+#####################################################################
+# Generate EC2 Key Pair for log in access to EC2
+#####################################################################
+
+resource "tls_private_key" "ssh_key" {
+  count     = var.ssh_key_name == "" ? 1 : 0
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "aws_key_pair" "aws_ssh_key" {
+  count      = var.ssh_key_name == "" ? 1 : 0
+  key_name   = "ec2-key-pair-${random_id.testing_id.hex}"
+  public_key = tls_private_key.ssh_key[0].public_key_openssh
+}
+
+locals {
+  ssh_key_name        = var.ssh_key_name != "" ? var.ssh_key_name : aws_key_pair.aws_ssh_key[0].key_name
+  private_key_content = var.ssh_key_name != "" ? var.ssh_key_value : tls_private_key.ssh_key[0].private_key_pem
+}
```

Comment:
Why do we need to make a key pair in terraform? We already use a PEM file for SSH access to the hosts.

Reply:
We make this to provide the option for dev on-board with integration-test does not needed to setup their resources (e.g only add another test case without worrying about set up a SSH Key). However, they can still provide their PEM file to ssh into the host.